components/extensions/vscode/src/Commands.ts

```diff
@@ -42,106 +45,149 @@ class Commands {
      * cellery build command handler used by the function 'registerCommand'
      */
     private static readonly handleBuildCommand = async() => {
-        const cellName = await vscode.window.showInputBox({
+        const errorMessage: string = "Unable to run cellery build";
+        let imageName = await vscode.window.showQuickPick(Commands.imageNames, {
             placeHolder: `${Constants.ORG_NAME}/${Constants.IMAGE_NAME}:${Constants.VERSION}`,
-            prompt: `Enter the image name`,
         });
-        const file = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.fileName : null;
-        if (cellName && file) {
-            const buildCommand = `${Constants.CELLERY_BUILD_COMMAND} ${file} ${cellName}`;
-            if (!(Commands.terminals.build)) {
-                Commands.terminals.build = vscode.window.createTerminal({
-                    name: "Cellery Build",
-                    cwd: path.dirname(file),
-                });
-            }
-            Commands.terminals.build.show(true);
-            Commands.terminals.build.sendText(buildCommand);
-        } else {
-            const errorMessage: string = "Unable to run cellery build";
-            if (!file) {
-                vscode.window.showErrorMessage(`${errorMessage}, source file not found`);
-            } else if (!cellName) {
-                vscode.window.showErrorMessage(`${errorMessage}, image name not found`);
+        if (imageName === undefined) {
+            vscode.window.showErrorMessage(`${errorMessage}, image name not found`);
+            return;
+        }
+        if (imageName === Commands.imageNames[0]) {
+            const newImageName = await CommandsUtils.getNewImageName();
+            if (newImageName === undefined || !CommandsUtils.validateImageTag(newImageName)) {
+                if (newImageName === undefined) {
+                    vscode.window.showErrorMessage(`${errorMessage}, image name not found`);
+                }
+                return;
             }
+            Commands.imageNames.push(newImageName);
+            imageName = newImageName;
+        }
+        const file = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.document.fileName : null;
+        if (file === null) {
+            vscode.window.showErrorMessage(`${errorMessage}, source file not found`);
```

Comment:
Let's update this message to be informative for the user. Please do so for similar errors.
We can also add `console.log()` or `console.error()` in the relevant places for logging purposes.

Suggestion:- Failed to run cellery build as no Cell file open. Please open a Cell file to perform this action.